tests/autorestart/test_container_autorestart.py

```diff
@@ -55,42 +55,42 @@ def ignore_expected_loganalyzer_exception(loganalyzer):
-        ".*ERR monit.*",
-    ]
-    swss_ignoreRegex = [
-        ".*ERR swss#orchagent.*removeLag.*",
-    ]
-    pmon_ignoreRegex = [
-        ".*ERR pmon#xcvrd.*initializeGlobalConfig.*",
-        ".*ERR pmon#thermalctld.*Caught exception while initializing thermal manager.*",
-    ]
-    syncd_ignoreRegex = [
-        ".*ERR syncd#syncd.*driverEgressMemoryUpdate.*",
-        ".*ERR syncd#syncd.*brcm_sai*",
-        ".*WARNING syncd#syncd.*saiDiscover: skipping since it causes crash.*",
-    ]
-    teamd_ignoreRegex = [
-        ".*ERR swss#portsyncd.*readData.*netlink reports an error=-33 on reading a netlink socket.*",
-    ]
-    systemd_ignoreRegex = [
-        ".*ERR systemd.*Failed to start .* container*",
-    ]
-    kernel_ignoreRegex = [
-        ".*ERR kernel.*PortChannel.*",
-    ]
-    other_ignoreRegex = [
-        ".*ERR route_check.*",
-    ]
+    swss_syncd_teamd_regex = [
+            ".*ERR swss#orchagent.*removeLag.*",
+            ".*ERR syncd#syncd.*driverEgressMemoryUpdate.*",
+            ".*ERR syncd#syncd.*brcm_sai*",
+            ".*ERR syncd#syncd.*SAI_API_UNSPECIFIED:sai_api_query.*",
+            ".*ERR syncd#syncd.*SAI_API_SWITCH:sai_query_attribute_enum_values_capability.*",
+            ".*ERR syncd#syncd.*SAI_API_SWITCH:sai_object_type_get_availability.*",
+            ".*ERR syncd#syncd.*sendApiResponse: api SAI_COMMON_API_SET failed in syncd mode.*",
+            ".*ERR syncd#syncd.*processQuadEvent.*",
+            ".*WARNING syncd#syncd.*skipping since it causes crash.*",
+            ".*ERR swss#portsyncd.*readData.*netlink reports an error=-33 on reading a netlink socket.*",
+            ".*ERR swss#orchagent.*set status: SAI_STATUS_ATTR_NOT_IMPLEMENTED_0.*",
+            ".*ERR swss#orchagent.*setIntfVlanFloodType.*",
+        ]
+    ignore_regex_dict = {
+        'common' : [
+            ".*ERR monit.*",
+            ".*ERR systemd.*Failed to start .* container*",
+            ".*ERR kernel.*PortChannel.*",
+            ".*ERR route_check.*",
+        ],
+        'pmon' : [
+            ".*ERR pmon#xcvrd.*initializeGlobalConfig.*",
+            ".*ERR pmon#thermalctld.*Caught exception while initializing thermal manager.*",
+        ],
+        'swss' : swss_syncd_teamd_regex,
+        'syncd' : swss_syncd_teamd_regex,
+        'teamd' : swss_syncd_teamd_regex,
+    }
+
+    _, feature = decode_dut_port_name(enum_dut_feature)
+
     if loganalyzer:
-        loganalyzer.ignore_regex.extend(monit_ignoreRegex)
-        loganalyzer.ignore_regex.extend(swss_ignoreRegex)
-        loganalyzer.ignore_regex.extend(pmon_ignoreRegex)
-        loganalyzer.ignore_regex.extend(syncd_ignoreRegex)
-        loganalyzer.ignore_regex.extend(teamd_ignoreRegex)
-        loganalyzer.ignore_regex.extend(systemd_ignoreRegex)
-        loganalyzer.ignore_regex.extend(kernel_ignoreRegex)
-        loganalyzer.ignore_regex.extend(other_ignoreRegex)
+        loganalyzer.ignore_regex.extend(ignore_regex_dict['common'])
```

Comment:
A minor comment, will the common set of match regex be added multiple times when ignore_expected_loganalyzer_exception() is called per enum_dut_feature ?

Reply:
Ignore my comment Ying .. I think it should be ok as it is parameterized and will be invoked now separately per enum_dut_feature.

Reply:
I think we are fine. the loganalyzer is setup per testcase.